src/arch.rs

```diff
@@ -10,26 +12,25 @@ pub enum Arch {
 }
 
 #[cfg(unix)]
-/// Get a sane default architecture for the platform.
-pub fn default_str() -> &'static str {
+/// handle common case: Apple Silicon / Node < 16
+pub fn get_safe_arch<'a>(arch: &'a Arch, version: &Version) -> &'a Arch {
     use crate::system_info::{platform_arch, platform_name};
 
-    // TODO: Handle (arch, name, version) when Node v15+ supports darwin-arm64
-    match (platform_name(), platform_arch()) {
-        ("darwin", "arm64") => "x64",
-        (_, arch) => arch,
-    }
+    return match (platform_name(), platform_arch(), version) {
+        ("darwin", "arm64", Version::Semver(v)) if v.major < 16 => &Arch::X64,
```

Comment:
Technically clobbers `--arch` options for darwin/arm.

I suppose we can early return for `Arch::Default.default() != arch`

Reply:
Although, I'd be interested in hearing a use case for that!...

Reply:
I think the only use case for that would be to allow custom archs, like `musl`, which was requested here: #239